### PR TITLE
Added Lesson08 Circle.py and Revised mailroom part 2 and mailroom part 3

### PR DESCRIPTION
**Pull request by Chieu1**

### CODE REVIEW

File: students/chieu_quach/lesson08/circle.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        self.radius = radius
+
+    @property
+    def get_diameter(self):
+   # def get_diameter(self, value):
+      
+        return self.radius * 2   
+
+    @get_diameter.setter
+    def get_diameter(self, value):
```
Comment (natasha-aleksandrova) [inline students/chieu_quach/lesson08/circle.py]: note that in directions this was called `diameter` (properties are usually nouns)

File: students/chieu_quach/lesson08/circle.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+    def __rmul__(self):
+        
+        return self.__mul__(other)
+
+    def __repr__(self):
+        return repr(self.radius)
+
+    def __str__(self):
+        return str(self.radius)
```
Comment (natasha-aleksandrova) [inline students/chieu_quach/lesson08/circle.py]: the idea with `__str__` and `__repr__` is that you customize what gets printed out

File: students/chieu_quach/lesson08/circle.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    def __eq__(self, other):
+        
+        if self.radius == other.radius:
+            return True
+        else:
+            return False
+
+    
+    
+    name = property(get_diameter, area)
```
Comment (natasha-aleksandrova) [inline students/chieu_quach/lesson08/circle.py]: Step 5 is missing? `from_diameter` alternate constructor?

File: students/chieu_quach/lesson08/circle.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+    # Multiply two Cicle values (use __mul__  and __rmul)
+    def __mul__(self, other):
+
+        total_num = self.radius * 3
+        return Circle(total_num)
+
+    def __rmul__(self):
+        
+        return self.__mul__(other)
```
Comment (natasha-aleksandrova) [inline students/chieu_quach/lesson08/circle.py]: where is `other` coming from?

File: students/chieu_quach/lesson08/circle_test.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+def test_add():
+    c  = Circle(2)
+    c2 = Circle(3)
+    assert repr(c + c2) == '5'
+
+def test_mult():
+    c = Circle(2)
+    num = 3
+    assert repr(c * num) == '6'
```
Comment (natasha-aleksandrova) [inline students/chieu_quach/lesson08/circle_test.py]: your test should assert on class attribute values instead of using `repr()` method.

**Comment by natasha-aleksandrova** [comment]
@Chieu1 I'm unable to merge due to merge conflicts. please make sure to merge with upstream and then we can see if it resolves the issues.

**Comment by Chieu1** [comment]
Natasha,

Per your comment, I have updated Circles exercise from lesson 8 to include step 5. I did PR on circle exercise for you to review. Please let me know if you see it.

For the mailroom 2 and 3 exercises,
I have corrected the indentation and resubmitted earlier. Can you let me know if you see them ? 
Previously,  I got 13 out of 15 for those two exercises because they have indentation violation.

Thanks,

Chieu

File: students/chieu_quach/lesson08/circle.py [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+
+    @get_diameter.setter
+    def get_diameter(self, value):
+        self.radius = value
+
+    @classmethod
+    def from_diameter(cls, radius):
+        cls.radius = radius / 2
+        
+        return cls(cls.radius)
```
Comment (natasha-aleksandrova) [inline students/chieu_quach/lesson08/circle.py]: this is pretty close...


should be:
```
@classmethod
    def from_diameter(cls, diameter):
        return cls(diameter / 2.0)
```

**Comment by natasha-aleksandrova** [comment]
per my comment before, I cannot merge this due to conflicts - you need to merge with upstream.

**Comment by Chieu1** [comment]
Natasha 

Thank you very much for your help.

You are a very good instructor. 

I remembered that I had problem with the sort method in mailroom exercise because I used dictionary of dictionaries method. I could not sort in ascendind order. I called you at Zoom meeting and you told me to re-work on the dictionary. At that time, I was mad at you.  I am really sorry about that. I did what you suggested and was able to sort it in ascending order. If you had not provided me with that suggestion, I don’t think I would complete mailrooms 3 and 4 and pass this course.

Again, thank you very much for your help!

Appreciate your help.

I will try to complete lesson 9 and submit for review by either Wednesday ir Thursday

Thanks,

Chieu

**Comment by natasha-aleksandrova** [comment]
I'm really happy you were able to get it working!

**Comment by Chieu1** [comment]
Natasha and Charles,

It has been my pleasure to have you as my Instructor and Tutor for this python course. 

I’ve learned  a lot from this challenging course. At time I thought I would not complete and pass this course. With your help to guide me through it and to allow me to have two weeks extended time, I am able to complete and pass this course.

Thanks Natasha for taking your time to provide comments on my gradings and to assist me with my homework at Zoom meeting.

Thanks Charles to assist me with my homework and to provide comments on my gradings.

Please help me to answer this question, Will I receive anything after I pass this course ? I need to turn in something to my company HR in order to get  tuition reimbursement from them.

Again, thank you very much Natasha and Charles 

Chieu

**Comment by natasha-aleksandrova** [comment]
@Chieu1 send us an email and we can provide additional information to you privately.

**Comment by Chieu1** [comment]
Natasha,

This is my email chieu.quach@yahoo.com

Thank you very much for your help.

It was nice to have you as my Instructor. I hope to apply my learning skill in python in the workforce.

Thanks,

Chieu

**Comment by natasha-aleksandrova** [comment]
@Chieu1 just to clarify - send _us_ an email message with your request. This is a public space.

**Comment by Chieu1** [comment]
Natasha,

I am requesting that you send me information about my passing grade in python course. Please send me information including paid tuition fee, course name, and passing grade comfirmation

Thanks

Chieu

**Comment by Chieu1** [comment]
Natasha,

I will send email message to the University to request for paid tuition fee and python passing grade info.

Thanks,

Chieu

**Pull request closed without merging**